scripts/lib/LinkChecker.test.ts

```diff
@@ -0,0 +1,159 @@
+    let testLink = new Link("/testpath", ["/testorigin.mdx"]);
+    let testFile = new File("docs/testpath.mdx");
+    let allFiles: File[] = [testFile];
+    const results = testLink.checkLink(allFiles);
+    expect(results).toEqual([true]);
+  });
+  test("Validate non-existing internal links with absolute path", () => {
+    let testLink = new Link("/test-alternative-path", ["/testorigin.mdx"]);
+    let testFile = new File("docs/testpath.mdx");
+    let allFiles: File[] = [testFile];
+    const results = testLink.checkLink(allFiles);
+    expect(results).toEqual([false]);
+  });
+  test("Validate existing internal links with relative path", () => {
+    let testLink = new Link("../testpath", ["docs/test/testorigin.mdx"]);
+    let testFile = new File("docs/testpath.mdx");
+    let allFiles: File[] = [testFile];
+    const results = testLink.checkLink(allFiles);
+    expect(results).toEqual([true]);
+  });
+  test("Validate non-existing internal links with relative path", () => {
+    let testLink = new Link("../testpath", ["docs/testorigin.mdx"]);
+    let testFile = new File("docs/testpath.mdx");
+    let allFiles: File[] = [testFile];
+    const results = testLink.checkLink(allFiles);
+    expect(results).toEqual([false]);
+  });
+  test("Validate existing internal links with absolute path and multiple origin files", () => {
+    let testLink = new Link("/testpath", [
+      "docs/test/testorigin.mdx",
+      "docs/test/test2/testorigin.mdx",
+      "docs/test/test3/testorigin.mdx",
+      "docs/test/test2/test4/testorigin.mdx",
+    ]);
+    let testFile1 = new File("docs/testpath.mdx");
+    let testFile2 = new File("docs/test/test2/testpath.mdx");
+    let allFiles: File[] = [testFile1, testFile2];
+    const results = testLink.checkLink(allFiles);
+    expect(results).toEqual([true, true, true, true]);
+  });
+  test("Validate non-existing internal links with absolute path and multiple origin files", () => {
+    let testLink = new Link("/testpath", [
+      "docs/test/testorigin.mdx",
+      "docs/test/test2/testorigin.mdx",
+      "docs/test/test3/testorigin.mdx",
+      "docs/test/test2/test4/testorigin.mdx",
+    ]);
+    let testFile1 = new File("docs/test/testpath.mdx");
+    let testFile2 = new File("docs/test2/test3/testpath.mdx");
+    let allFiles: File[] = [testFile1, testFile2];
+    const results = testLink.checkLink(allFiles);
+    expect(results).toEqual([false, false, false, false]);
+  });
+  test("Validate internal links with relative path and multiple origin files", () => {
+    let testLink = new Link("../testpath", [
+      "docs/test/testorigin.mdx",
+      "docs/test/test2/testorigin.mdx",
+      "docs/test/test3/testorigin.mdx",
+      "docs/test/test2/test4/testorigin.mdx",
+    ]);
+    let testFile1 = new File("docs/testpath.mdx");
+    let testFile2 = new File("docs/test/test2/testpath.mdx");
+    let allFiles: File[] = [testFile1, testFile2];
+    const results = testLink.checkLink(allFiles);
```

Comment:
Nit: inlining is simpler

```suggestion
    const results = testLink.checkLink([testFile1, testFile2]);
```

Ditto the other tests